spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -104,50 +104,107 @@ <h1>Runtime Semantics: BackreferenceMatcher ( _n_, _direction_ )</h1>
+    1. Return _obj_.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-tomatchoptions" aoid="ToMatchOptions">
+    <h1><ins>ToMatchOptions ( _Obj_ )</ins></h1>
+    <p>
+      When the abstract operation ToMatchOptions is called with Object _Obj_, the following steps are taken:
+    </p>
+    <emu-alg>
+    1. If _Obj_ is *undefined*, then
+      1. Let _optionsObj_ be ObjectCreate(*null*).
+    1. Else,
+      1. Let _optionsObj_ be ? ToObject(_options_).
+    1. Let _capture_ be ? GetOption(_options_, `"capture"`, `"string"`, &laquo; `"strings"`, `"indices"` &raquo;, `"strings"`).
+    1. Return a new MatchOptions { [[Capture]]: _capture_ }.
+    </emu-alg>
+  </emu-clause>
+  </ins>
+</emu-clause>
+
 <emu-clause id="sec-properties-of-the-regexp-prototype-object">
   <h1>Properties of the RegExp Prototype Object</h1>
 
   <emu-clause id="sec-regexp.prototype.exec">
-    <h1>RegExp.prototype.exec ( _string_ <ins>[, _mapfn_ ]</ins> )</h1>
+    <h1>RegExp.prototype.exec ( _string_ <ins>[, _options_ ]</ins> )</h1>
     <p>Performs a regular expression match of _string_ against the regular expression and returns an Array object containing the results of the match, or *null* if _string_ did not match.</p>
     <p>The String ToString(_string_) is searched for an occurrence of the regular expression pattern as follows:</p>
     <emu-alg>
     1. Let _R_ be the *this* value.
     1. If Type(_R_) is not Object, throw a *TypeError* exception.
     1. If _R_ does not have a [[RegExpMatcher]] internal slot, throw a *TypeError* exception.
-    1. <ins>If _mapfn_ is not *undefined* and IsCallable(_mapfn_) is *false*, throw a *TypeError* exception.</ins>
+    1. <ins>Let _matchOptions_ be ? ToMatchOptions(_options_).</ins>
     1. Let _S_ be ? ToString(_string_).
-    1. Return ? RegExpBuiltinExec(_R_, _S_<ins>, _mapfn_</ins>).
+    1. Return ? RegExpBuiltinExec(_R_, _S_<ins>, _matchOptions_</ins>).
     </emu-alg>
 
     <emu-clause id="sec-regexpexec" aoid="RegExpExec">
-      <h1>Runtime Semantics: RegExpExec ( _R_, _S_<ins>, _mapfn_</ins> )</h1>
-      <p>The abstract operation RegExpExec with arguments <del>_R_ and _S_</del><ins>_R_, _S_, and _mapfn_</ins> performs the following steps:</p>
+      <h1>Runtime Semantics: RegExpExec ( _R_, _S_<ins>, _matchOptions_</ins> )</h1>
+      <p>The abstract operation RegExpExec with arguments <del>_R_ and _S_</del><ins>_R_, _S_, and MatchOptions _matchOptions_</ins> performs the following steps:</p>
       <emu-alg>
       1. Assert: Type(_R_) is Object.
       1. Assert: Type(_S_) is String.
-      1. <ins>Assert: _mapfn_ is *undefined* or IsCallable(_mapfn_) is *true*.</ins>
       1. Let _exec_ be ? Get(_R_, `"exec"`).
       1. If IsCallable(_exec_) is *true*, then
-        1. Let _result_ be ? Call(_exec_, _R_, &laquo; _S_<ins>, _mapfn_</ins> &raquo;).
+        1. Let _result_ be ? Call(_exec_, _R_, &laquo; _S_<ins>, ! FromMatchOptions(_matchOptions_)</ins> &raquo;).
```

Comment:
this is an observable breaking change, since this would call subclass exec methods.

Reply:
Any existing code using RegExp subclasses that does *not* use `{ captures: "indices" }` would not be significantly affected, as the default behavior is used if `undefined` is passed. The only observable break would be code using a RegExp subclass that chooses to use `{ captures: "indices" }`. If the subclass is not updated to support this argument, they would get an array of strings as per the default behavior.

Reply:
ah, my bad, it's only a break from your earlier spec :-) not from 262 itself.